source/m2mbase.cpp

```diff
@@ -442,12 +511,21 @@ bool M2MBase::is_integer(const String &value)
 
 void M2MBase::set_uri_path(const String &path)
 {
-    _uri_path = path;
+    free(_uri_path);
+    _uri_path = NULL;
```

Comment:
This NULL assignment is pointless, as the alloc_copy() return NULL one out-of-memory case anyway.